test/server/cards/03_TWI/units/EncouragingLeadership.spec.ts

```diff
@@ -0,0 +1,51 @@
+describe('Encouraging Leadership', function () {
+    integration(function (contextRef) {
+        it('should give +1/+1 for this phase to allies when played', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['wampa', 'encouraging-leadership'],
+                    spaceArena: ['green-squadron-awing'],
+                    groundArena: ['duchesss-champion', 'atst']
+                },
+                player2: {
+                    spaceArena: ['star-wing-scout'],
+                    groundArena: ['specforce-soldier'],
+                },
+            });
+            const { context } = contextRef;
+
+            // Check if effect is applied to both units
+            context.player1.clickCard(context.encouragingLeadership);
+            expect(context.encouragingLeadership).toBeInZone('discard');
+
+            // Check if effect is applied to allies
+            expect(context.atst.getPower()).toEqual(7);
+            expect(context.atst.getHp()).toEqual(8);
+            expect(context.greenSquadronAwing.getPower()).toEqual(2);
+            expect(context.greenSquadronAwing.getHp()).toEqual(4);
+
+            // Check if effect is not applied to non-allies
+            expect(context.starWingScout.getPower()).toEqual(4);
+            expect(context.starWingScout.getHp()).toEqual(1);
+            expect(context.specforceSoldier.getPower()).toEqual(2);
+            expect(context.specforceSoldier.getHp()).toEqual(2);
+
+            // Check if effect is not applied to card played after the Encouraging Leadership
+            context.player2.passAction();
+            context.player1.clickCard(context.wampa);
+            expect(context.wampa.getPower()).toEqual(4);
+            expect(context.wampa.getHp()).toEqual(5);
+
+            // Pass the phase
+            context.moveToRegroupPhase();
+
+            // Check if units power/hp is back to normal
+            expect(context.greenSquadronAwing.getPower()).toEqual(1);
+            expect(context.atst.getPower()).toEqual(6);
+            expect(context.atst.getHp()).toEqual(7);
+            expect(context.greenSquadronAwing.getPower()).toEqual(1);
+            expect(context.greenSquadronAwing.getHp()).toEqual(3);
```

Comment:
Apologies for the late request but I just thought of one more potential bug it would be great to have a check for - could you add a third unit that receives the buff and is then defeated while the buff is still active?

The targeting for the stat buff effects has been a little glitchy so we just need to make sure it doesn't cause an issue by trying to apply itself you the unit after defeat.

Reply:
I'm not sure to follow. You want me to add another unit on my board that gets defeated ? which check would you expect to have for this unit in particular ?

Reply:
Yep that's it, thanks!